src/bufferedBot.ts

```diff
@@ -0,0 +1,58 @@
+import TelegramBot from 'node-telegram-bot-api'
+import { Subject, timer } from 'rxjs'
+import { bufferTime, filter, groupBy, mergeMap, concatMap, ignoreElements, startWith } from 'rxjs/operators'
+
+import { SendMessageInput, concatMessages } from 'helpers'
+import { Logger } from '@gnosis.pm/dex-js'
+
+const log = new Logger('bot:buffered')
+
+const BUFFER_TIME = 3000 // consecutive messages over this time get buffered
+const SPACE_TIME = 1000 // min time between sending messages
+
+// sendMessage buffers messages and sends them in batches
+export class BufferedBot extends TelegramBot {
+  private messagesSubject$ = new Subject<SendMessageInput>()
+
+  constructor(token: string, options?: TelegramBot.ConstructorOptions) {
+    super(token, options)
+
+    this.messagesSubject$.asObservable().pipe(
```

Comment:
This code is a poem, we should frame it. Thanks for the comments, I think they are useful